### PR TITLE
Fix handling artifacts defined in modules

### PR DESCRIPTION
**Pull request by goldmann**
* Fixed typo for resource description
* Description is displayed when handling of resource fails
* Fixed handling of 'path' resources with cache enabled

### CODE REVIEW

File: concreate/descriptor.py [hunk 1]
```diff
@@ -104,6 +106,21 @@ def _process_ports(self):
...
+
+        for artifact in artifacts:
+            if 'name' not in artifact:
+                # This is suboptimal, but at this point we cannot reuse created
+                # resource object. We do create it only to retrieve the name
+                # of the artifact. We do this because we can have different types
+                # of resources (path, url, etc).
+                artifact['name'] = Resource.new(artifact, self.directory).name
+
+        self.descriptor['artifacts'] = artifacts
```
Comment (dbecvarik) [inline concreate/descriptor.py]: you are putting Resources classes to descriptor

Comment (goldmann) [reply concreate/descriptor.py]: I'm not :)

Comment (goldmann) [reply concreate/descriptor.py]: And yes, if you have a better/cleaner idea - please propose one, I couldn't find one.

Comment (dbecvarik) [reply concreate/descriptor.py]: whats wrong with old way just - create Resource Object when is needed and not store it in descriptor?
https://github.com/goldmann/concreate/blob/fe7faf84d9e8daa68f253080ea9b179deb0a7182/concreate/generator.py#L82

Comment (goldmann) [reply concreate/descriptor.py]: It's dona at different time for modules and main images - thus - we don't have 'name' populated which fails. This was the main reason for this PR.

Comment (dbecvarik) [reply concreate/descriptor.py]: So create name here and reuse it later - name should be responsibility of descriptor and resource should only consume it and do not touch it.

Comment (goldmann) [reply concreate/descriptor.py]: Yes, we can move responsibility for creating name from resource to descriptor.

**Comment by goldmann** [comment]
@dbecvarik What's the decision here? IIRC on IRC you agreed to approve it, correct?

File: concreate/descriptor.py (hunk 1 above)
Comment (rcernich) [reply concreate/descriptor.py]: Default name generation is resource specific, which is why I put the code in the specific resource objects.  Notice the git resource uses a different algorithm for calculating its default name.  The effect of this is that there ends up being a single "if" block for determining the resource type and all subsequent specialization is done by the class.

**Pull request merged**